GVFS/GVFS.Common/ProductUpgrader.cs

```diff
@@ -216,6 +216,35 @@ public virtual bool TryCreateToolsDirectory(out string upgraderToolPath, out str
             return false;
         }
 
+        public bool TryCleanup(out string error)
+        {
+            bool isError = false;
+            string exceptionInfo = string.Empty;
+            foreach (Asset asset in this.newestRelease.Assets)
+            {
+                Exception exception;
+                if (!this.TryDeleteDownloadedAsset(asset, out exception))
+                {
+                    exceptionInfo += $"Could not delete {asset.LocalPath}. {exception.ToString()}.";
+                    isError = true;
+                }
+            }
+
+            if (isError)

```

Comment:
Do you need a second variable for this or can you just check if `exceptionInfo` is non-empty?

Reply:
Related: do you need exceptionInfo, or can use just use `error`?

Reply:
I can use just the error.